server/client_server.go

```diff
@@ -52,3 +129,99 @@ func (csc *confluenceServerClient) GetSelf() (*types.ConfluenceUser, error) {
+		}
+	}
+
+	if strings.Contains(webhookPayload.Event, Page) {
+		confluenceServerEvent.Page, err = csc.GetPageData(int(webhookPayload.Page.ID))
+		if err != nil {
+			return nil, errors.Errorf("error getting page data for the event. PageID %d. Error: %v", webhookPayload.Page.ID, err)
+		}
+	}
+
+	if strings.Contains(webhookPayload.Event, Space) {
+		confluenceServerEvent.Space, err = csc.GetSpaceData(webhookPayload.Space.SpaceKey)
+		if err != nil {
+			return nil, errors.Errorf("error getting space data for the event. SpaceKey %s. Error: %v", webhookPayload.Space.SpaceKey, err)
+		}
+	}
+
+	return &confluenceServerEvent, nil
+}
+
+func (csc *confluenceServerClient) GetCommentData(webhookPayload *serializer.ConfluenceServerWebhookPayload) (*CommentResponse, error) {
+	commentResponse := &CommentResponse{}
+	if _, err := service.CallJSONWithURL(csc.URL, fmt.Sprintf("%s%s?expand=body.view,container,space,history", PathCommentData, strconv.FormatInt(webhookPayload.Comment.ID, 10)), http.MethodGet, nil, commentResponse, csc.HTTPClient); err != nil {
+		return nil, err
+	}
+
+	commentResponse.Body.View.Value = util.GetBodyForExcerpt(commentResponse.Body.View.Value)
+
+	return commentResponse, nil
+}
+
+func (csc *confluenceServerClient) GetPageData(pageID int) (*PageResponse, error) {
+	pageResponse := &PageResponse{}
+	if _, err := service.CallJSONWithURL(csc.URL, fmt.Sprintf("%s%s?status=any&expand=body.view,container,space,history", PathPageData, strconv.Itoa(pageID)), http.MethodGet, nil, pageResponse, csc.HTTPClient); err != nil {
+		return nil, err
+	}
+
+	pageResponse.Body.View.Value = util.GetBodyForExcerpt(pageResponse.Body.View.Value)
+
+	return pageResponse, nil
+}
+
+func (csc *confluenceServerClient) GetSpaceData(spaceKey string) (*SpaceResponse, error) {
+	spaceResponse := &SpaceResponse{}
+	if _, err := service.CallJSONWithURL(csc.URL, fmt.Sprintf("%s%s?status=any", PathSpaceData, spaceKey), http.MethodGet, nil, spaceResponse, csc.HTTPClient); err != nil {
+		return nil, err
+	}
+
+	return spaceResponse, nil
+}
+
+type apiResponse struct {
+	Results []struct {
+		ID   int64  `json:"id"`
+		Key  string `json:"key"`
+		Name string `json:"name"`
+	} `json:"results"`
+	Size int `json:"size"`
+}
+
+func (csc *confluenceServerClient) GetSpaceKeyFromSpaceID(spaceID int64) (string, error) {
+	start := 0
+
+	for {
+		path := fmt.Sprintf("%s?start=%d&limit=%d", PathAllSpaces, start, pageSize)
+
+		response := &apiResponse{}
+
+		if _, err := service.CallJSONWithURL(csc.URL, path, http.MethodGet, nil, response, csc.HTTPClient); err != nil {
+			return "", errors.Wrap(err, "confluence GetSpaceKeyFromSpaceID")
+		}
+
+		for _, space := range response.Results {
+			if space.ID == spaceID {
+				return space.Key, nil
+			}
+		}
+
+		if len(response.Results) < pageSize {
+			break
+		}
+
+		start += pageSize
+	}
+
+	return "", fmt.Errorf("confluence GetSpaceKeyFromSpaceID: no space found for the space key")
```

Comment:
Should this error say, "no space key found for the space id" ?

Reply:
Yes, that should have been. Fixed it now
Updated it to `confluence GetSpaceKeyFromSpaceID: no space key found for the space id`